include/experimental/__p0009_bits/macros.hpp

```diff
@@ -101,6 +103,52 @@
 #define MDSPAN_IMPL_STANDARD_NAMESPACE_STRING MDSPAN_PP_STRINGIFY(MDSPAN_IMPL_STANDARD_NAMESPACE)
 #define MDSPAN_IMPL_PROPOSED_NAMESPACE_STRING MDSPAN_PP_STRINGIFY(MDSPAN_IMPL_STANDARD_NAMESPACE) "::" MDSPAN_PP_STRINGIFY(MDSPAN_IMPL_PROPOSED_NAMESPACE)
 
+namespace MDSPAN_IMPL_STANDARD_NAMESPACE {
+namespace detail {
+
+inline void default_precondition_violation_handler(const char* cond, const char* file, unsigned line)
+{
+  (void)std::fprintf(::stderr, "%s:%u: precondition failure: `%s`\n", file, line, cond);
+  std::abort();
+}
```

Comment:
I don't know how HIP or SYCL work, but the following should be fine for CUDA.  Note that in device code, `printf` may have a performance cost even if it is never called at run time.  (For example, the compiler may need to reserve registers for it.)

```suggestion
{
#if defined(__CUDA_ARCH__)
#  if ! defined(NDEBUG)
  printf("%s:%u: precondition failure: `%s`\n", file, line, cond);
  assert(false);
#  endif
#else
  fprintf(stderr, "%s:%u: precondition failure: `%s`\n", file, line, cond);
  assert(false);
#endif 
}
```

Reply:
`HIP` works the same as `Cuda`. For SYCL (oneAPI), you would need to use `sycl::ext::oneapi::experimental::printf`.